src/simulation_strategies/GeneralizedCliffordsSimulationStrategy.py

```diff
@@ -44,7 +46,7 @@ def __get_sorted_possible_states(self) -> None:
         possible_input_states = self.__calculate_all_input_substates(self.input_state[:])
 
         # Labeling them into dict where keys are being number of particles in the state.
-        self._labeled_states = dict()
+        self._labeled_states = defaultdict()
```

Comment:
You use `defaultdict` but without the factory. Didn't you want each entry to be a list? Then `default_dict(list)` would provide you with a `list` under each key that was not present in the dictionary.